### PR TITLE
Break pkg/logs/input into launchers and tailers

### PR DESCRIPTION
**Pull request by djmitche**
### What does this PR do?

Breaks `pkg/logs/input`, which was a mix of tailers and launchers (defined generously) into `pkg/logs/internal/launchers` and `pkg/logs/internal/tailers`.

### Motivation

This is one in a series of updates to the logs agent to make it simpler and more maintainable.

In this case: launchers and tailers are different things, and although they are sort-of "one-to-one" right now, that may not be the case in the future -- for example, we might have a "container" launcher that works with any runtime, but creates runtime-specific tailers.  And even if they were one-to-one, a conceptual separation between the two eases the reader's comprehension.

### Additional Notes

In light of team discussion today, I'd like to highlight that this PR is "big" in the sense that it has lots of lines changed, but "small" in the sense that it doesn't make any functional changes and is fairly formulaic.  I'll follow up with more PRs that may make minor functional changes, but those should be more easily reviewed individually.  If I have made any functional changes, please let me know!

This should be reviewed commit-by-commit.  Many are completely formulaic, and should be quick to skim.  Some are a _little_ more complex, and the comments for those commits will guide your understanding of the changes.

### Possible Drawbacks / Trade-offs

If you had `pkg/logs/input/file/scanner.go` programmed into your fingers, you'll need to reprogram your fingers.  But hopefully the new organization is simple enough to figure out.

### Describe how to test/QA your changes

No need.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] The appropriate `team/..` label has been applied, if known.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/logs/internal/tailers/file/tailer.go [hunk 1]
```diff
@@ -20,38 +20,33 @@ import (
...
+	// File contains the logs configuration for the File to parse (path, source, ...)
+	// If you are looking for the os.File use to read on the FS, see osFile.
+	File *File
 
 	fullpath string
 	osFile   *os.File
 	tags     []string
 
-	outputChan  chan *message.Message
+	OutputChan  chan *message.Message
```
Comment (gh123man) [inline pkg/logs/internal/tailers/file/tailer.go:42]: were these properties made public just to satisfy unit tests?

File: pkg/logs/internal/launchers/file/launcher.go [hunk 2]
```diff
@@ -38,9 +38,9 @@ var ContainersLogsDir = "/var/log/containers"
 // DefaultSleepDuration represents the amount of time the tailer waits before reading new data when no data is received
 const DefaultSleepDuration = 1 * time.Second
 
-// Scanner checks all files provided by fileProvider and create new tailers
+// Launcher checks all files provided by fileProvider and create new tailers
 // or update the old ones if needed
-type Scanner struct {
+type Launcher struct {
```
Comment (gh123man) [inline pkg/logs/internal/launchers/file/launcher.go:43]: I agree that file `Launcher` is a better name given it's use case. 
We will have to update docs including the readme you also have a PR open (and probably the more recent slide decks) for to account for this for consistency.

File: pkg/logs/internal/tailers/file/tailer.go (hunk 1 above)
Comment (djmitche) [reply pkg/logs/internal/tailers/file/tailer.go:42]: I think so.  We can clean that up in the followup PRs.

File: pkg/logs/internal/launchers/file/launcher.go (hunk 2 above)
Comment (djmitche) [reply pkg/logs/internal/launchers/file/launcher.go:43]: I merged that README here and updated it, so landing those in the correct order should work.  I'm not terribly comfortable editing other people's presentations, though.

File: pkg/logs/decoder/file_decoder.go [hunk 3]
```diff
@@ -0,0 +1,54 @@
+package decoder
```
Comment (ogaca-dd) [inline pkg/logs/decoder/file_decoder.go:1]: Missing copyrights?

**Review by ogaca-dd — APPROVED** [review]
Great PR, very easy to review 💯 (and not big by the way)

File: pkg/logs/internal/tailers/file/tailer.go (hunk 1 above)
Comment (gh123man) [reply pkg/logs/internal/tailers/file/tailer.go:42]: ok just don't want to lose track of it.

**Review by gh123man — APPROVED** [review]

**Comment by djmitche** [comment]
I'm not sure why @DataDog/container-integrations is included in this PR.  @DataDog/infrastructure-integrations is included because that team owns the traps launcher and tailer.  I just made a CODEOWNERS push to update that path as well.

**Review by AlexandreYang — APPROVED** [review]

**Review by AlexandreYang — APPROVED** [review]
👍  for snmp traps related files

**Review by FlorianVeaux — APPROVED** [review]
Looks good for file owned by the infrastructure-integrations team

**Comment by djmitche** [comment]
(I suspect container-integrations was here because of the merge from main)

**Pull request merged**